### ISSUE TITLE
KeepAlive agent options typo

### ISSUE THREAD

**Issue by calvinwiebe** [elastic/elasticsearch-js#582]
Hey @spalger,

There's been some recent discussion on #521, but it seems to have gotten lost in the backlog. :P. There seems to be a couple of issues being brought up in that discussion, so I wanted to separate out one in particular. Could you take a look at my [comment](https://github.com/elastic/elasticsearch-js/issues/521#issuecomment-314869540) with regards to the keep alive agent timeout options. 

In short, it seems like `elasticsearch-js` is using `agentkeepalive ^2.2.0`, which expects `keepAliveTimeout`, but `elasticsearch-js` is passing it `freeSocketKeepAliveTimeout`. I've remedied this locally for myself by overriding the `createNodeAgent` option to pass in `keepAliveTimeout` to the `agentkeepalive` constructor. Legit? Looking for some closure on this :). Thanks!

**Comment by bkw** [comment]
+100

**Comment by shaharmor** [comment]
Anyone?

**Comment by spalger** [comment]
Fixed in #600
